File: examples/stream/listMessages/go/listMessages.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/stream"
)

// List messages for a given channel
func ListMessages() {
	streamService := stream.NewStreamService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := streamService.ListMessages(&stream.ListMessagesRequest{
		Channel: "general",
	})
	fmt.Println(rsp, err)
}

File: examples/stream/listMessages/node/listMessages.js
const { StreamService } = require("micro-js-client/stream");

// List messages for a given channel
async function listMessages() {
  let streamService = new StreamService(process.env.MICRO_API_TOKEN);
  let rsp = await streamService.listMessages({
    channel: "general",
  });
  console.log(rsp);
}

listMessages();
